### ISSUE TITLE
A11y: windows terminal emits selection/deselection events in mark mode when navigating with arrow keys

### ISSUE THREAD

**Issue by beqabeqa473** [microsoft/terminal#13447]
### Windows Terminal version

1.15.1862.0

### Windows build number

10.0.19044.1766

### Other Software

- Narrator
- NVDA latest version

### Steps to reproduce

- Run Narrator on NVDA;
- Open Windows terminal;
- Go to mark mode with ctrl+shift+m;
- Navigate text with arrow keys.

### Expected Behavior

Screen readers should announce only characters when navigating with arrow keys.

### Actual Behavior

Screen readers announce selection of the current char plus unselection of previous char in addition with current char announcement

**Comment by codeofdusk** [comment]
CC @carlos-zamora.

**Comment by carlos-zamora** [comment]
Hmm... This seems to be operating as expected. When you enter mark mode we create a single-cell selection at the current cursor position. The arrow keys then move that single-cell selection as described [in the docs](https://docs.microsoft.com/en-us/windows/terminal/selection#keyboard-support). Since we're selecting an adjacent character and unselecting the one we were just on, everything seems to be working as expected.

That said, we do want to get rid of that single-cell selection and instead move the cursor (just like in conhost). It's doable, but it requires a bit of refactoring under the hood. Doing so would require...
- selections to be stored as an inclusive start and an exclusive end (similar to how UIA handles it). Currently they're stored as inclusive start and end.
- ensure we differentiate between no selection and an empty selection (particularly important when copying text and calling `IsSelectionActive`)
- (from the UIA text provider side) `GetSelection()` needs to be able to get any of the following: (1) the selected range, (2) an empty selection exposed as the cursor, and (3) the cursor when no selection exists. We already do (1) and (3). (2) just requires us to be careful the previous point.

I'll chat about it with Dustin today to create a game plan here.

**Comment by beqabeqa473** [comment]
It needs a refactor indeed, because in a way how it is working now it
is not so comfortable to use this feature.

On 7/7/22, Carlos Zamora ***@***.***> wrote:
> Hmm... This seems to be operating as expected. When you enter mark mode we
> create a single-cell selection at the current cursor position. The arrow
> keys then move that single-cell selection as described [in the
> docs](https://docs.microsoft.com/en-us/windows/terminal/selection#keyboard-support).
> Since we're selecting an adjacent character and unselecting the one we were
> just on, everything seems to be working as expected.
>
> That said, we do want to get rid of that single-cell selection and instead
> move the cursor (just like in conhost). It's doable, but it requires a bit
> of refactoring under the hood. Doing so would require...
> - selections to be stored as an inclusive start and an exclusive end
> (similar to how UIA handles it). Currently they're stored as inclusive start
> and end.
> - ensure we differentiate between no selection and an empty selection
> (particularly important when copying text and calling `IsSelectionActive`)
> - (from the UIA text provider side) `GetSelection()` needs to be able to get
> any of the following: (1) the selected range, (2) an empty selection exposed
> as the cursor, and (3) the cursor when no selection exists. We already do
> (1) and (3). (2) just requires us to be careful the previous point.
>
> I'll chat about it with Dustin today to create a game plan here.
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/microsoft/terminal/issues/13447#issuecomment-1178003681
> You are receiving this because you authored the thread.
>
> Message ID: ***@***.***>


-- 
with best regards Beqa Gozalishvili
Tell: +995593454005
Email: ***@***.***
Web: https://gozaltech.org
Skype: beqabeqa473
Telegram: https://t.me/gozaltech
facebook: https://facebook.com/gozaltech
twitter: https://twitter.com/beqabeqa473
Instagram: https://instagram.com/beqa.gozalishvili

**Comment by carlos-zamora** [comment]
Since this is a decent size change, we probably won't service this to 1.15, so I'm removing it from the servicing pipeline.
That said, I'll be working to fix this up to work as expected. Currently scheduled for 1.16 (next release).

**Comment by beqabeqa473** [comment]
@carlos-zamora are there any news in this area?
